src/components/Puzzles/DailyPuzzle/FetchDailyPuzzle.tsx

```diff
@@ -0,0 +1,49 @@
+import { useEffect } from "react";
+import { Puzzle } from "./PuzzleTypes";
+import { doc, setDoc, getDoc } from "firebase/firestore";
+import { db } from "../../../firebase/firebase.ts";
+
+interface FetchDailyPuzzleProps {
+  setPuzzle: (puzzle: Puzzle) => void;
+}
+
+const FetchDailyPuzzle = ({ setPuzzle }: FetchDailyPuzzleProps) => {
```

Comment:
O widzę, że jeden komentarz mi się nie zatwierdził ostatnio :)

To powinien być hook, a nie komponent, który zwraca null :)

Reply:
Jak to, nie rozumiem za bardzo

Reply:
https://react.dev/learn/reusing-logic-with-custom-hooks